### ISSUE TITLE
data-match validator not monitoring target field for changes

### ISSUE THREAD

**Issue by Jargon64** [1000hz/bootstrap-validator#316]
Think I've discovered a bug with the data-match validator. Take the example at http://jsbin.com/hisoha/6/edit?html,js,console,output

If you provide a valid email address in the first field, then retype the same email address in the second the validation works as expected. However, if you then go back to the first email field and enter in a completely different, yet valid, email address the validation still succeeds and the form can be submitted.

I assume this is due to the way the valid state is cached per field for optimisation purposes (after reading through other issues on this board).

The only way I can safely proceed is to put a data-match on both email fields (referencing the other), but then the form starts whining at you before you even get to the confirm email field.

**Comment by 1000hz** [comment]
Hmm, yeah. I'll investigate.

**Comment by marcelbonnet** [comment]
Hey, great project. I don't wanna be the one to echo "hey I found the same bug", but I was trying to implement a equlasgreater and equalslesser using the same approach of the "match" function and than I realized the same bug then happens to my new functions. So it would be really great to fix the mentioned bug.

**Comment by 1000hz** [comment]
It's a result of the recent change that skips a field's validation if the input hasn't been touched. It needs to be smarter about it. I probably won't have time to fix it very soon.

**Comment by alytle** [comment]
Confirmed, I reverted to the previous release and the problem was fixed.

**Comment by Jargon64** [comment]
I've run into this issue a second time in a different project while creating a custom validator that depends on another field. Decided to take a look into the source code and saw that the cached result check and assignment is done before the validators are run. So what I ended up doing is clearing the cached result at the beginning of my custom validator knowing it will force the validator to run on the next check. The code I added to my validator is as follow:

`$el.removeData('bs.validator.previous');`

Hopefully this work around helps some people out.
